File: COCOA iOS/appPickerView2Columnas/appPickerView2Columnas/ViewController.m
//
//  ViewController.m
//  appPickerView2Columnas
//
//  Created by Axel Roman on 17/04/24.
//

#import "ViewController.h"

@interface ViewController ()
{
    NSArray *productos;
    NSArray *colores;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    productos = @[@"Pantalla LCD",@"iPad",@"Bicicleta",@"Motocicleta",@"Carro",@"Camioneta"];
    colores = @[@"Rojo 🌶",@"Verde 🐸",@"Azul 🐬",@"Gris 👽",@"Naranja 🍊",@"Aleatorio 🔀"];
    
    _pickerView1.dataSource = self;
    _pickerView1.delegate = self;
    
    _label1.text = [NSString stringWithFormat:@"%@, %@",[productos objectAtIndex:0],[colores objectAtIndex:0]];
    
    NSLog(@"%@", _label1.text);
    
    UIColor*color = [UIColor colorWithRed:0 green:175 blue:150 alpha:1];
    
    self.view.backgroundColor = color;
    
    _imageView1.backgroundColor = [UIColor colorWithRed:161 green:133 blue:148 alpha:1];
    
    _imageView1.image = [UIImage imageNamed:@"PantallaLCD"];
 
    
}

-(NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView{
    return 2;
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component{
    if(component == 0)
        return productos.count;
    else
        return colores.count;
}

- (NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component{
    if(component == 0)
        return [productos objectAtIndex:row];
    else
        return colores[row];
    return nil;
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component{
    _label1.text = [NSString stringWithFormat:@"%@, %@", [productos objectAtIndex:[_pickerView1 selectedRowInComponent:0]], [colores objectAtIndex:[_pickerView1 selectedRowInComponent:1]]];
    
    int color = (int) [colores indexOfObject:[colores objectAtIndex:[_pickerView1 selectedRowInComponent:1]]];
    
    switch (color) {
        case 0:
            _imageView1.backgroundColor = [UIColor redColor];
            break;
            
        case 1:
            _imageView1.backgroundColor = [UIColor greenColor];
            break;
            
        case 2:
            _imageView1.backgroundColor = [UIColor blueColor];
            break;
            
        case 3:
            _imageView1.backgroundColor = [UIColor grayColor];
            break;
            
        case 4:
            _imageView1.backgroundColor = [UIColor orangeColor];
            break;
            
        case 5:
            srand((unsigned int) time(NULL));
            UIColor * color =[UIColor colorWithRed:rand()%255/255.0 green:rand()%255/255.0 blue:rand()%255/255.0 alpha:1];
            
            _imageView1.backgroundColor = color;
            break;
            
    }
    
    int producto = (int) [productos indexOfObject:[productos objectAtIndex:[_pickerView1 selectedRowInComponent:0]]];
    
    switch (producto) {
        case 0:
            _imageView1.image = [UIImage imageNamed:@"PantallaLCD"];
            break;
        
        case 1:
            _imageView1.image = [UIImage imageNamed:@"ipad"];
            break;
            
        default:
            break;
    }
}


@end

File: iOS Swift/appPickerViewProductos/appPickerViewProductos/ViewController.swift
//
//  ViewController.swift
//  appPickerViewProductos
//
//  Created by Axel Roman on 27/05/24.
//

import UIKit

class ViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource {
    
    let carro_color = [["VW","Toyota","Ford","Chrysler","Honda","Mazda","RAM"],["🚘","🚕","🚗"]]
    
    let imagenPorOpcion: [String: String] = [
        "VW": "VW polo",
        "Toyota": "TOYOTA",
        "Ford": "FORD1",
        "Chrysler": "chrysler",
        "Honda": "honda",
        "Mazda": "mazda1",
        "RAM": "ramn1"
    ]

//    let imagenPorOpcion = [["VW","TOYOTA","FORD1","BICI_1","moto1","mazda","camioneta1"]]

    @IBOutlet weak var label1: UILabel!
    
    @IBOutlet weak var texto1: UITextField!
    
    @IBOutlet weak var pickerView1: UIPickerView!
    
    @IBOutlet weak var imagenes: UIImageView!
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        pickerView1.delegate = self
        pickerView1.dataSource = self
        
    }
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        return 2
    }
    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return carro_color[component].count
    }
    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
        return carro_color[component][row]
    }
    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        let carro = carro_color[0][pickerView.selectedRow(inComponent: 0)]
        let color = carro_color[1][pickerView.selectedRow(inComponent: 1)]
        
        label1.text = carro + " " + color
        texto1.text = carro + " " + color
        
       // imagenes.image = UIImage(named: imagenPorOpcion[row])
        
        if let imagen = imagenPorOpcion[carro_color[0][pickerView.selectedRow(inComponent: 0)]] {
                    imagenes.image = UIImage(named: imagen)
                }
    }


}

